src/image/loading_displaying.js

```diff
@@ -361,7 +361,8 @@ p5.prototype.saveGif = async function(
 
   // initialize variables for the frames processing
   let frameIterator = nFramesDelay;
-  this.frameCount = frameIterator;
+  // BUGFIX: Removed line that was resetting frameCount
```

Comment:
We actually intentionally set the frame count here so that sketches relying on the frame count are effectively reset when recording the gif. (An extension we haven't got to yet is to also override millis() when recording too.) Fully removing it would break the use case where you want to get a recording from the start of the animation.

Maybe there's a way the user can pass an option into saveGif to tell it to pick up recording from whatever the current state of the sketch is, and we don't touch frameCount (and in the future, other timing state) when that's set?